demoAndroid/src/main/java/com/novoda/demoandroid/returnvalue/ReturnValueActivity.java

```diff
@@ -0,0 +1,79 @@
+package com.novoda.demoandroid.returnvalue;
+
+import android.annotation.SuppressLint;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.novoda.demoandroid.R;
+import com.novoda.demoandroid.SecondLevelBaseActivity;
+import com.novoda.sexp.ElementFinderInstigator;
+import com.novoda.sexp.RootTag;
+import com.novoda.sexp.SimpleEasyXmlParser;
+import com.novoda.sexp.finder.ElementFinder;
+import com.novoda.sexp.finder.ElementFinderFactory;
+
+public class ReturnValueActivity extends SecondLevelBaseActivity {
+    // language=XML
+    private static final String XML = "<novoda>"
+            + "<favouriteColour>Blue</favouriteColour>" + "</novoda>";
+
+    private TextView parsingResult;
+    private ProgressBar progressBar;
+    private LinearLayout container;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_result_parsing);
+
+        parsingResult = (TextView) findViewById(R.id.tv_result);
+        progressBar = (ProgressBar) findViewById(R.id.pb_oneTag);
+        container = (LinearLayout) findViewById(R.id.ll_oneTag);
+
+        ElementFinderFactory factory = SimpleEasyXmlParser.getElementFinderFactory();
+        ElementFinder<String> elementFinder = factory.getStringFinder();
+        SimpleInstigator instigator = new SimpleInstigator(elementFinder, "favouriteColour");
+
+        new ParsingTask(XML, instigator).execute();
+    }
+
+    private static class SimpleInstigator extends ElementFinderInstigator<String> {
```

Comment:
(I don't fully understand the reason for this class either) but my two pennies:

[Instigate](http://dictionary.reference.com/browse/instigate)
> to urge, provoke, or incite to some action or course:

dunno if it is a metaphor, an Instigator is something that Instigates, something that starts it, so here you need an Instigator to Instigate parsing of the XML.

But all the names are poor, i.e. here I would more argue about `Simple` .. but this is done because it's a demo.

Same with `ElementFinderInstigator` I wasn't actual sure what to call it, and our other is called `SimpleTagInstigator` which is another poor name.

So I agree, poor naming all round!